### PR TITLE
Made robot_controller agnostic to rospy.

### PR DESCRIPTION
**Pull request by hdavidzhu**

### CODE REVIEW

File: comprobo_final_project/scripts/robot_controller.py [hunk 1]
```diff
@@ -65,28 +55,13 @@ def position_callback(self, msg):
...
-                r.sleep()
-            except rospy.exceptions.ROSTimeMovedBackwardsException:
-                print "Time went backwards. Carry on."
+        # Set linear and angular velocities
+        self.robot.set_twist(forward_rate, turn_rate)
 
 
 if __name__ == '__main__':
     robot_controller = RobotController([0.0, 1.0, 2.0, 3.0, 4.0, 5.0])
-    robot_controller.run()
```
Comment (kzhang8850) [inline comprobo_final_project/scripts/robot_controller.py]: What's the 1 for?

File: comprobo_final_project/scripts/robot_controller.py [hunk 2]
```diff
@@ -65,28 +55,13 @@ def position_callback(self, msg):
...
-        self.pub.publish(cmd_vel)
-
-        
-    def run(self):
-        """
-        Main run function.
-        """
-
-        r = rospy.Rate(10)
+        forward_rate = a1*diff_w + b1*diff_r + c1*diff_r**2
```
Comment (kzhang8850) [inline comprobo_final_project/scripts/robot_controller.py]: how does this work? no more run function?

File: comprobo_final_project/scripts/robot_controller.py [hunk 3]
```diff
@@ -65,28 +55,13 @@ def position_callback(self, msg):
...
+        turn_rate = a2*diff_w + b2*diff_r + c2*diff_r**2
 
-        while not rospy.is_shutdown():
-            # Check for time-jumps, like when looping a bag file
-            try:
-                r.sleep()
-            except rospy.exceptions.ROSTimeMovedBackwardsException:
-                print "Time went backwards. Carry on."
+        # Set linear and angular velocities
+        self.robot.set_twist(forward_rate, turn_rate)
```
Comment (kzhang8850) [inline comprobo_final_project/scripts/robot_controller.py]: do you not publish anymore?

File: comprobo_final_project/scripts/models/robot.py [hunk 4]
```diff
@@ -1,2 +1,44 @@
...
+from ..providers.gazebo_position_provider import GazeboPoseProvider
+
 class Robot:
-    pass
+    """
+    Neato connection.
+    set_twist(twist : Twist) : Void - Sets the Twist of the robot
+    """
+
+    def __init__(self, position_listener):
```
Comment (kzhang8850) [inline comprobo_final_project/scripts/models/robot.py]: i'm fairly confused by the interactions between this file and the other. I will currently just assume that they work.

File: comprobo_final_project/scripts/robot_controller.py (hunk 3 above)
Comment (hdavidzhu) [reply comprobo_final_project/scripts/robot_controller.py]: Publishing happens in `Robot`, inside of `robot.set_twist`

File: comprobo_final_project/scripts/robot_controller.py (hunk 1 above)
Comment (hdavidzhu) [reply comprobo_final_project/scripts/robot_controller.py]: 1 second

**Review by PrimePalaver — APPROVED** [review]

**Pull request merged**